File: query/src/main/java/com/lantanagroup/nandina/query/pihc/fhir/r4/cerner/report/PatientReportModel.java
package com.lantanagroup.nandina.query.pihc.fhir.r4.cerner.report;

public class PatientReportModel {
  private String facilityId;
  private String censusId;
  private String patientId;
  private String admitDate;
  private String dischargeDate;
  private String age;
  private String sex;
  private String race;
  private String ethnicity;
  private String chiefComplaint;
  private String primaryDx;
  private String location;
  private String disposition;

  public String getDisposition() {
    return disposition;
  }

  public void setDisposition(String disposition) {
    this.disposition = disposition;
  }

  public String getLocation() {
    return location;
  }

  public void setLocation(String location) {
    this.location = location;
  }

  public String getPrimaryDx() {
    return primaryDx;
  }

  public void setPrimaryDx(String primaryDx) {
    this.primaryDx = primaryDx;
  }

  public String getChiefComplaint() {
    return chiefComplaint;
  }

  public void setChiefComplaint(String chiefComplaint) {
    this.chiefComplaint = chiefComplaint;
  }

  public String getEthnicity() {
    return ethnicity;
  }

  public void setEthnicity(String ethnicity) {
    this.ethnicity = ethnicity;
  }

  public String getRace() {
    return race;
  }

  public void setRace(String race) {
    this.race = race;
  }

  public String getSex() {
    return sex;
  }

  public void setSex(String sex) {
    this.sex = sex;
  }

  public String getAge() {
    return age;
  }

  public void setAge(String age) {
    this.age = age;
  }

  public String getDischargeDate() {
    return dischargeDate;
  }

  public void setDischargeDate(String dischargeDate) {
    this.dischargeDate = dischargeDate;
  }

  public String getAdmitDate() {
    return admitDate;
  }

  public void setAdmitDate(String admitDate) {
    this.admitDate = admitDate;
  }

  public String getPatientId() {
    return patientId;
  }

  public void setPatientId(String patientId) {
    this.patientId = patientId;
  }

  public String getCensusId() {
    return censusId;
  }

  public void setCensusId(String censusId) {
    this.censusId = censusId;
  }

  public String getFacilityId() {
    return facilityId;
  }

  public void setFacilityId(String facilityId) {
    this.facilityId = facilityId;
  }
}

File: web/src/main/angular/app/report-body/pillbox-report/patient-report-model.ts
import {BaseReportModel} from './base-report-model';

export class PatientReportModel extends BaseReportModel {
  facilityId: string;
  censusId: string;
  patientId: string;
  admitDate: string;
  dischargeDate: string;
  age: string;
  sex: string;
  race: string;
  ethnicity: string;
  chiefComplaint: string;
  primaryDx: string;
  location: string;
  disposition: string;

  constructor(data?: any) {
    super();
    if (data) Object.assign(this, data);
  }

  get patientIdCode() {
    return this.getCodeDisplay(this.patientId);
  }

  get patientIdSystem() {
    return this.getCodeSystem(this.patientId);
  }

  get raceCode() {
    return this.getCodeDisplay(this.race);
  }

  get raceSystem() {
    return this.getCodeSystem(this.race);
  }

  get ethnicityCode() {
    return this.getCodeDisplay(this.ethnicity);
  }

  get ethnicitySystem() {
    return this.getCodeSystem(this.ethnicity);
  }

  get chiefComplaintCode() {
    return this.getCodeDisplay(this.chiefComplaint);
  }

  get chiefComplaintSystem() {
    return this.getCodeSystem(this.chiefComplaint);
  }

  get primaryDxCode() {
    return this.getCodeDisplay(this.primaryDx);
  }

  get primaryDxSystem() {
    return this.getCodeSystem(this.primaryDx);
  }

  get dispositionCode() {
    return this.getCodeDisplay(this.disposition);
  }

  get dispositionSystem() {
    return this.getCodeSystem(this.disposition);
  }
}